util/tracing/tracer.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracing
+
+import (
+	"context"
+
+	"github.com/opentracing/basictracer-go"
+	"github.com/opentracing/opentracing-go"
+)
+
+// Snowball is set as Baggage on traces which are used for snowball tracing.
+const Snowball = "sb"
+
+// A CallbackRecorder immediately invokes itself on received trace spans.
+type CallbackRecorder func(sp basictracer.RawSpan)
+
+// RecordSpan implements basictracer.SpanRecorder.
+func (cr CallbackRecorder) RecordSpan(sp basictracer.RawSpan) {
+	cr(sp)
+}
+
+// NewSnowball returns a Span which records directly via the specified
+// callback. If the given WireSpan is the zero value, a new trace is created;
+// otherwise, the created Span is a child.
+func NewSnowball(opName string, callback func(sp basictracer.RawSpan)) opentracing.Span {
+	tr := basictracer.New(CallbackRecorder(callback))
+	opentracing.SetGlobalTracer(tr)
```

Comment:
It seems `NewSnowBall()` will be called in `TraceExecutor.Open` which will be called per statement.

So If two people do trace command same time with two connections, this global variable will be a race condition and be modified in two different goroutines?

Reply:
It is by design. Only `TracExec` calls `NewSnowBall()`. It will be evolving quickly, so I prefer to keeping it like this.

Reply:
plz explain more about the correctness of this by design~~~?

multi goroutines(os threads underneath) write/read same gobal variable seems be a well-known bug. (ref: https://golang.org/ref/mem#tmp_1). 

I still can not understand how can u got right value without any memory corruption in multi trace stmt exec.

Reply:
By design means there will be only one `TracExec` running. In general, this feature is executed from MySQL-shell and usually done by ops engineer or client. We just need to collect useful information.

Reply:
got it~